internal/controller/policyexceptiondraft_controller.go

```diff
@@ -69,118 +76,164 @@ func (r *PolicyExceptionDraftReconciler) Reconcile(ctx context.Context, req ctrl
 		namespace = r.DestinationNamespace
 	}
 
-	// Check if the draft hast been reconciled before
-	if _, exists := exceptionDraft.Labels["kyverno-policy-operator/reconciled"]; !exists {
-		// Label doesn't exist, add it
-		if exceptionDraft.Labels == nil {
-			exceptionDraft.Labels = make(map[string]string)
+	// Create Kyverno exception
+	// Create a policy map for storing draft policies to extract rules later
+	policyMap := make(map[string]kyvernov1.ClusterPolicy)
+	for _, policy := range exceptionDraft.Spec.Policies {
+		var kyvernoPolicy kyvernov1.ClusterPolicy
+		if err := r.Get(ctx, types.NamespacedName{Namespace: "", Name: policy}, &kyvernoPolicy); err != nil {
```

Comment:
Future improvement: don't fetch policies on every reconciliation loop

Reply:
Was thinking about this. What if the Policies get modified?

Reply:
You can populate a cache and update it if/when policies are updated. There are builtins for doing this, see the starboard-exporter cache for the service endpoints